internal/controller/postgrescluster/pgmonitor.go

```diff
@@ -160,12 +157,9 @@ func (r *Reconciler) reconcileMonitoringSecret(
 		return nil, err
 	}
 
-	if !pgmonitor.ExporterEnabled(cluster) {
-		// TODO: Checking if the exporter is enabled to determine when monitoring
-		// secret should be created. If more tools are added to the monitoring
-		// suite, they could need the secret when the exporter is not enabled.
-		// This check may need to be updated.
-		// Exporter is disabled; delete monitoring secret if it exists.
+	// Checking if the exporter is enabled to determine when monitoring
+	// secret should be created.
```

Comment:
:pick: 
```suggestion
	// Checking if the exporter is enabled or OpenTelemetryMetrics feature
	// is enabled to determine when monitoring secret should be created,
	// since our implementation of the SqlQuery receiver in the OTel Collector
	// uses the monitoring user as well.
```